scripts/install_apt_deps.sh

```diff
@@ -1,21 +1,92 @@
 #!/bin/bash
 
+
+### Initial setup ###
+
+# Source environment variables describing the current OS distribution
 . /etc/os-release
 
+
+### Package dependencies setup ###
+
+# List of common package dependencies for the ILLIXR project
+pkg_dep_list_common="
```

Comment:
Curious, any reason behind the ordering here?

Reply:
I tried not to change it too much but I tried to follow a few rules:
- If a set of packages are logically grouped, I put them on the same line. May have missed some groups due to unfamiliarity of when/why they were added.
- General dev dependencies like git I tried putting at the top.
- Specialized dev dependencies (e.g. lib.*-dev) and new dependencies at the bottom. These can be broken out for specific features/dependencies (a la realsense)  .
- Further rearrangement can be made with some hints as to what should be grouped.

Reply:
Yeah, that sounds reasonable. Groupings are tricky and its easy to lose track of what is what. But let's take a crack at the easy ones. Add comments in the script to name the groups for future reference.

Display / window manager: `libx11-xcb-dev libx11-dev libxcb-glx0-dev libxcb-randr0-dev libxrandr-dev libxkbcommon-dev libwayland-dev libsdl2-dev`

You already have the "mesa", "GL", and "USB" groups. Should we make one each for threading and linear algebra?

Reply:
And maybe also an "essentials" or "general" group with stuff like git, cmake, etc.